Packs/DigitalShadows/Integrations/ReliaQuestGreyMatterDRPIncidents/ReliaQuestGreyMatterDRPIncidents.py

```diff
@@ -0,0 +1,1011 @@
+        # a greater granularity of information, such as the credential-exposure endpoint which contains the actual
+        # credential we have found exposed in a specific field on the model
+        alerts = self.get_alerts(alert_triage_items=alert_triage_items)
+
+        incident_ids = {ti[SOURCE][INCIDENT_ID] for ti in triage_items if
+                        INCIDENT_ID in ti[SOURCE] and ti[SOURCE][INCIDENT_ID]}
+        incidents = get_incidents(self.request_handler, incident_ids=incident_ids)
+
+        asset_ids = {
+            asset[ID] for alert_or_incident in [*alerts, *incidents] for asset in alert_or_incident[ASSETS]}
+        assets = get_assets(self.request_handler, asset_ids=asset_ids)
+        return self.merge_data([], triage_items, triage_item_comments, alerts, incidents, assets, True)
+
+
+''' FETCH INCIDENT '''
+
+
+def fetch_incidents(fetchLimit, ingestClosed, riskLevel, riskTypes, search_light_request_handler, sinceDate):
+    """
+    fetch incidents will take config for fetching and ingesting incidents in xsoar
+    Args:
+        fetchLimit: no of incidents needs to fetch per iteration
+        ingestClosed: closed incidents should be ingested or not
+        riskLevel: risk levels needs to ingest
+        riskTypes: risk types needs to ingest
+        search_light_request_handler: request handler to fetch data
+        sinceDate: since when we want to ingest data
+    """
+    last_run = demisto.getLastRun()
+    last_event_num = last_run.get('incidents', {}).get('last_fetch', 0)
+    LOG(f"fetch_incidents last run: {last_event_num}")
+    search_list_triage_poller = SearchLightTriagePoller(search_light_request_handler)
+    poll_result = search_list_triage_poller.poll_triage(event_num_start=last_event_num, limit=fetchLimit,
+                                                        event_created_after=sinceDate,
+                                                        alert_risk_types=riskTypes, risk_level=riskLevel,
+                                                        should_ingest_closed=ingestClosed)
+    data = poll_result.triage_data
+    last_polled_number = poll_result.max_event_number
+    if data:
+        incidents = []
+        for item in data:
+            incident = {
+                'name': item["triage_item"]['title'],
+                'occurred': item["triage_item"]['raised'],
+                'rawJSON': json.dumps(item)
+            }
+            incidents.append(incident)
+        demisto.incidents(incidents)
+        LOG(f"settings  last run: {last_event_num}")
+        demisto.setLastRun({'incidents': {'last_fetch': last_polled_number}})
+        LOG(f"data found for iteration last_polled_number:{last_polled_number}")
+    else:
+        LOG(f"No data found for iteration last_polled_number:{last_polled_number}")
+        demisto.incidents([])
+        LOG(f"settings  last run: {last_event_num}")
+        demisto.setLastRun({'incidents': {'last_fetch': last_polled_number}})
```

Comment:
Can be cleaned up a bit:

```suggestion
    if data:
        incidents = [
            {
                'name': item["triage_item"]['title'],
                'occurred': item["triage_item"]['raised'],
                'rawJSON': json.dumps(item)
            }
            for item in data
        ]
        LOG(f"data found for iteration {last_polled_number=}")
    else:
        incidents = []
        LOG(f"No data found for iteration {last_polled_number=}")
    
    demisto.incidents(incidents)
    LOG(f"setting last run: {last_event_num}")
    demisto.setLastRun({'incidents': {'last_fetch': last_polled_number}})
```

Reply:
sure will do that